### PR TITLE
Feat(/#265):- Back to top component.

### PR DESCRIPTION
**Pull request by anjneeksharma**
# Closes/Fixes/Resolves

### 

# Explain the feature/fix
It is back to top web component based on lit element that can be used in other application, its main functionality to take user in starting of screen or top of screen.

## Does this PR introduce a breaking change

No

## Screenshots

<details>
<summary>View Screenshots</summary>

image

</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Does the change have appropriate unit tests?
- [x] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] My changes generate no new warnings
- [x] I have performed a self-review of my own code
- [x] I have documented my code, particularly in areas like todo, complex logic, quick fix, temporary patch, etc.
#### If it is a new component
- [x] Does your component follow One Platform style guidelines?
- [x] Does your component web accessibility standards? [Helper Doc](https://www.w3.org/TR/wai-aria-1.1/)
- [ ] Does your component support desktop screen sizes (350px, 720px, 1150px ,1920px)
#### Browsers you have tested in
- [ ] Latest two versions of Mozilla Firefox and Google Chrome supported by Red Hat Enterprise Linux distribution
- [x] Google Chrome [Latest 2 versions]
- [x] Mozilla Firefox [Latest 2 versions]
- [ ] Microsoft Edge [Latest 2 versions]
- [ ] Apple Safari [Latest 2 versions]
- [ ] Microsoft Internet Explorer 11

### CODE REVIEW

File: packages/opc-back-to-top/package.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{
+  "name": "@one-platform/opc-back-to-top",
+  "version": "0.0.1-prerelease",
+  "description": "A simple web component built in javascript and lit",
```
Comment (riginoommen) [inline packages/opc-back-to-top/package.json:4]: Please update the description.

File: packages/opc-back-to-top/test/opc-back-to-top.test.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import {LitElement} from 'lit-element';
```
Comment (riginoommen) [inline packages/opc-back-to-top/test/opc-back-to-top.test.ts:1]:
```suggestion
import { LitElement } from 'lit-element';
```

**Review by riginoommen — CHANGES_REQUESTED** [review]

**Comment by riginoommen** [comment]
Is the screenshot you shared is the correct design.

**Comment by anjneeksharma** [comment]
@riginoommen  yes thats as per design.

**Review by gisumit — CHANGES_REQUESTED** [review]
this PR contains changes that are not related to back to the top component. Can you update the branch and send the PR again.

File: packages/opc-back-to-top/package.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
+{
+  "name": "@one-platform/opc-back-to-top",
+  "version": "0.0.1-prerelease",
+  "description": "A web component built in lit element for Red Hat One Platform Back to top.",
```
Comment (riginoommen) [inline packages/opc-back-to-top/package.json:4]:
```suggestion
  "description": "Back to Top Component which enables smooth navigation from bottom scroll bar.",
```

**Review by riginoommen — CHANGES_REQUESTED** [review]

File: packages/opc-back-to-top/demo/index.html [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        background-color: #C0C0C0;
+    }
+     </style>
+</head>
+<body>
+    <div></div>
+        <opc-back-to-top></opc-back-to-top>
+        <opc-back-to-top>Back to Top</opc-back-to-top>
+        <opc-back-to-top reveal-at="200"></opc-back-to-top>
+        <opc-back-to-top reveal-at="300"></opc-back-to-top>
```
Comment (gisumit) [inline packages/opc-back-to-top/demo/index.html:20]: Can please add an example with a slot. and updated the screenshot & readme according to it.

**Review by gisumit — CHANGES_REQUESTED** [review]

File: packages/opc-back-to-top/demo/index.html (hunk 4 above)
Comment (anjneeksharma) [reply packages/opc-back-to-top/demo/index.html:20]: There is an example with slot also <opc-back-to-top>Back to Top</opc-back-to-top>  can you please elaborate what do you mean by slot ?

Comment (gisumit) [reply packages/opc-back-to-top/demo/index.html:20]: an example with slot.
```suggestion
        <opc-back-to-top reveal-at="300">
             <div>go to top</div>
        </opc-back-to-top>
```

Comment (anjneeksharma) [reply packages/opc-back-to-top/demo/index.html:35]: @gisumit  can you plz check.

**Comment by anjneeksharma** [comment]
@gisumit  plz have another look.

**Review by gisumit — APPROVED** [review]

**Review by hybridx — APPROVED** [review]

File: packages/opc-back-to-top/demo/index.html (hunk 4 above)
Comment (gisumit) [reply packages/opc-back-to-top/demo/index.html:35]: fixed by anjanee

File: packages/opc-back-to-top/README.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+plain back to top
+```html
+<opc-back-to-top></opc-back-to-top>
+```
+Back-to-top with reveal-at attribute, It reveal back to top once user scroll upt to specified px down.
+```html
+<opc-back-to-top reveal-at="200"></opc-back-to-top>
+```
+
+Back to top with slot, In slot user can pass its own choice icon and text.
```
Comment (ghanlohar) [inline packages/opc-back-to-top/README.md:25]:
```suggestion
Back to top with slot, In slot users can pass icon and text of their choice.
```

**Review by ghanlohar — CHANGES_REQUESTED** [review]

File: packages/opc-back-to-top/.npmignore [hunk 6]
```diff
@@ -0,0 +1,4 @@
+_temp
```
Comment (ghanlohar) [inline packages/opc-back-to-top/.npmignore:1]: Ignore config files as well. 
```suggestion
# Ignore config files
*.config.json

# Folders
_temp
```

**Review by ghanlohar — CHANGES_REQUESTED** [review]

**Review by ghanlohar — APPROVED** [review]

**Review by hybridx — APPROVED** [review]

File: packages/opc-back-to-top/README.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
+# This is opc-back-to-top 👋
```
Comment (diwanshi) [inline packages/opc-back-to-top/README.md:1]: rename to `This is opc-back-top-top component`

File: packages/opc-back-to-top/README.md [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+A standardized web component based on Lit element for Red Hat One Platform back to top.
+
+## Install
+
+```sh
+npm install
+```
+
+## Usage
+plain back to top
```
Comment (diwanshi) [inline packages/opc-back-to-top/README.md:16]: rename to `Default back-to-top`

File: packages/opc-back-to-top/README.md [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+```html
+<opc-back-to-top></opc-back-to-top>
```
Comment (diwanshi) [inline packages/opc-back-to-top/README.md:18]: @anjneeksharma this Readme doesn't follow the standard Readme format that we have created. Please update it accrodingly

Comment (diwanshi) [reply packages/opc-back-to-top/README.md:18]: all the attributes and slots should be mentioned separately

File: packages/opc-back-to-top/README.md [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+```
+Back-to-top with reveal-at attribute, It reveals back to top once user scrolls upto specified pixels downwards.
+```html
+<opc-back-to-top reveal-at="200"></opc-back-to-top>
+```
+
+Back to top with slot, In slot users can pass icon and text of their choice.
```
Comment (diwanshi) [inline packages/opc-back-to-top/README.md:25]: this should be written as `back-top-top with default slot (optional) where users can pass icon and text if they like`

File: packages/opc-back-to-top/package.json [hunk 11]
```diff
@@ -0,0 +1,47 @@
+{
+  "name": "@one-platform/opc-back-to-top",
+  "version": "0.0.1-prerelease",
+  "description": "Back to Top Component which enables smooth navigation from bottom scroll bar.",
```
Comment (diwanshi) [inline packages/opc-back-to-top/package.json:4]: rename to `Back to Top component that enables smooth scroll from the bottom of the page to the top of the page`

File: packages/opc-back-to-top/src/opc-back-to-top.ts [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+          Go to top
+      </button>
+   </slot>
+    
+    `;
+  }
+  goToTop() {
+    window.scrollTo(0, 0);
+  }
+  get toggleButton(){
```
Comment (diwanshi) [inline packages/opc-back-to-top/src/opc-back-to-top.ts:34]: all the getter functions and custom functions should come before the render function
here for eg: move get toggleButton() below static get styles() and goToTop() to just above the render function

File: packages/opc-back-to-top/src/opc-back-to-top.ts [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+    super();
+    window.addEventListener("scroll", ()=>{this.scrolledY=window.scrollY});
+  }
+
+  static get styles() {
+    return [ style ];
+  }
+  render() {
+    return html`
+    <slot @click="${this.goToTop}"  class="${this.toggleButton}">
```
Comment (diwanshi) [inline packages/opc-back-to-top/src/opc-back-to-top.ts:18]: there is an unnecessary space after `@click="${this.goToTop}"`

File: packages/opc-back-to-top/src/opc-back-to-top.ts [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+        <svg xmlns='http://www.w3.org/2000/svg' width='20px' height='20px' viewBox='0 0 512 512'>
+          <title>ionicons-v5-a</title>
+          <polyline points='112 244 256 100 400 244' style='fill:none;stroke:#000;stroke-linecap:round;stroke-linejoin:round;stroke-width:48px'/>
+            <line x1='256' y1='120' x2='256' y2='412' style='fill:none;stroke:#000;stroke-linecap:round;stroke-linejoin:round;stroke-width:48px'/>
+        </svg>
+          Go to top
+      </button>
+   </slot>
+    
+    `;
```
Comment (diwanshi) [inline packages/opc-back-to-top/src/opc-back-to-top.ts:29]: move this one on line27 after the closing slot

File: packages/opc-back-to-top/src/opc-back-to-top.scss [hunk 15]
```diff
@@ -0,0 +1,21 @@
+@import '@one-platform/opc-styles';
+
+$LOCAL: opc-back-to-top;
+
+$variables: ( 
+  //Add your variables here
+);
+
+.hide{
```
Comment (diwanshi) [inline packages/opc-back-to-top/src/opc-back-to-top.scss:9]: can u provide a space between the .hide class and opening curly brace i.e `.hide {`

File: packages/opc-back-to-top/src/opc-back-to-top.scss [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+  display: none;
+}
+.show {
+  font-family: 'Red Hat Text', sans-serif;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+button{
```
Comment (diwanshi) [inline packages/opc-back-to-top/src/opc-back-to-top.scss:18]: same here provide space bbetween class name and opening curly brace

File: packages/opc-back-to-top/src/opc-back-to-top.scss [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+
+.hide{
+  display: none;
+}
+.show {
+  font-family: 'Red Hat Text', sans-serif;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
```
Comment (diwanshi) [inline packages/opc-back-to-top/src/opc-back-to-top.scss:17]: add a line break after each class definition

File: packages/opc-back-to-top/test/opc-back-to-top.test.ts [hunk 18]
```diff
@@ -0,0 +1,39 @@
+import { LitElement } from 'lit-element';
+import { BackToTop } from '../src/opc-back-to-top';
+
+describe('opc-back-to-top', () => {
+
+    const OPC_COMPONENT = 'opc-back-to-top';
+    const ELEMENT_ID = 'opc-back-to-top';
+    let buttonElement: BackToTop;
+
+    const getShadowRoot = (tagName: string): ShadowRoot => {
```
Comment (diwanshi) [inline packages/opc-back-to-top/test/opc-back-to-top.test.ts:10]: do  not use getElementsbyTagName please follow the test suite structure as defined in header component.
You need to use querySelector

**Review by diwanshi — CHANGES_REQUESTED** [review]

File: packages/opc-back-to-top/README.md (hunk 7 above)
Comment (hybridx) [reply packages/opc-back-to-top/README.md:1]:
```suggestion
# This is opc-back-to-top  component 👋
```
@anjneeksharma  What @diwanshi mentioned (-top-top) must be a typo. I suppose she meant what I have suggested

File: packages/opc-back-to-top/src/opc-back-to-top.scss (hunk 17 above)
Comment (anjneeksharma) [reply packages/opc-back-to-top/src/opc-back-to-top.scss:17]: you mean there should be one line space between 2 classes. ?

File: packages/opc-back-to-top/README.md (hunk 9 above)
Comment (anjneeksharma) [reply packages/opc-back-to-top/README.md:18]: @diwanshi i am not able to understand your this comment, can you please elaborate bit ?

File: packages/opc-back-to-top/README.md (hunk 7 above)
Comment (diwanshi) [reply packages/opc-back-to-top/README.md:1]: yes that was a typo on my end

File: packages/opc-back-to-top/README.md (hunk 9 above)
Comment (diwanshi) [reply packages/opc-back-to-top/README.md:18]: @anjneeksharma can u use the standard readme format that we are using in opc-header component

File: packages/opc-back-to-top/src/opc-back-to-top.scss (hunk 17 above)
Comment (diwanshi) [reply packages/opc-back-to-top/src/opc-back-to-top.scss:17]: yes

Comment (anjneeksharma) [reply packages/opc-back-to-top/src/opc-back-to-top.scss:17]: Okay

**Comment by anjneeksharma** [comment]
@diwanshi  Please have a another look.

**Review by riginoommen — APPROVED** [review]

**Review by hybridx — APPROVED** [review]

File: packages/opc-back-to-top/README.md [hunk 19]
```diff
@@ -0,0 +1,131 @@
...
+
+## Run tests
+
+```sh
+npm run test
+```
+
+## 🤝 Contributors
+
+👤 **anjnee**
```
Comment (diwanshi) [inline packages/opc-back-to-top/README.md:131]: Add a proper name here with email.

File: packages/opc-back-to-top/src/opc-back-to-top.scss [hunk 20]
```diff
@@ -0,0 +1,24 @@
...
+
+button {
+  display: flex;
+
```
Comment (diwanshi) [inline packages/opc-back-to-top/src/opc-back-to-top.scss:22]: why have u added a space between properties?

File: packages/opc-back-to-top/src/opc-back-to-top.ts [hunk 21]
```diff
@@ -0,0 +1,38 @@
...
+  constructor() {
+    super();
+    window.addEventListener("scroll", ()=>{this.scrolledY=window.scrollY});
+  }
+
+  static get styles() {
+    return [ style ];
+  }
+
+  get toggleButton(){
```
Comment (diwanshi) [inline packages/opc-back-to-top/src/opc-back-to-top.ts:17]: provide a space between function name and opening braces

File: packages/opc-back-to-top/test/opc-back-to-top.test.ts [hunk 22]
```diff
@@ -0,0 +1,35 @@
...
+        buttonElement = window.document.createElement(OPC_COMPONENT) as BackToTop;
+        document.body.appendChild(buttonElement);
+    });
+
+    afterEach(() => {
+       document.querySelector(OPC_COMPONENT).remove();
+    });
+
+    it('is defined', async () => {
+        expect(buttonElement).toBeDefined();
```
Comment (diwanshi) [inline packages/opc-back-to-top/test/opc-back-to-top.test.ts:20]: fix your indentation

File: packages/opc-back-to-top/test/opc-back-to-top.test.ts [hunk 23]
```diff
@@ -0,0 +1,35 @@
...
+      });
+
+      it('has reveal-at defined', async () => {
```
Comment (diwanshi) [inline packages/opc-back-to-top/test/opc-back-to-top.test.ts:23]: fix indentation

File: packages/opc-back-to-top/test/opc-back-to-top.test.ts [hunk 24]
```diff
@@ -0,0 +1,35 @@
...
+        buttonElement.setAttribute('reveal-at', '200');
+        await buttonElement.updateComplete;
+        expect(buttonElement.revealAt).toEqual(200);
+      });
+
+      it('has reveal-at defined', async () => {
```
Comment (diwanshi) [inline packages/opc-back-to-top/test/opc-back-to-top.test.ts:29]: fix indentation

**Review by diwanshi — CHANGES_REQUESTED** [review]

File: packages/opc-back-to-top/test/opc-back-to-top.test.ts (hunk 22 above)
Comment (anjneeksharma) [reply packages/opc-back-to-top/test/opc-back-to-top.test.ts]: I followed select then right click then format document, is that same you follow to format your code ?

**Comment by anjneeksharma** [comment]
@diwanshi Please have a another look.

File: packages/opc-back-to-top/README.md [hunk 25]
```diff
@@ -0,0 +1,131 @@
...
+    <title>ionicons-v5-a</title>
+    <polyline points='112 328 256 184 400 328' style='fill:none;stroke:#000;stroke-linecap:round;stroke-linejoin:round;stroke-width:48px'/>
+    </svg>
+    Go-up
+  </button>
+</opc-back-to-top>
+```
+
+## Attributes
+<!-- Add attributes here -->
```
Comment (diwanshi) [inline packages/opc-back-to-top/README.md:44]: @anjneeksharma can u please add the attributes here pls

File: packages/opc-back-to-top/README.md [hunk 26]
```diff
@@ -0,0 +1,131 @@
...
+## Slots
+- Back-to-top with default slot (optional) where users can pass icon and text if they like.
+```html
+<opc-back-to-top>
+  <button class="rh-text" tabindex="0">
+    <svg xmlns='http://www.w3.org/2000/svg' width='20px' height='20px' viewBox='0 0 512 512'>
+    <title>ionicons-v5-a</title>
+    <polyline points='112 328 256 184 400 328' style='fill:none;stroke:#000;stroke-linecap:round;stroke-linejoin:round;stroke-width:48px'/>
+    </svg>
+    Go-up
```
Comment (diwanshi) [inline packages/opc-back-to-top/README.md:38]: rename this to `Go to Top`

**Review by diwanshi — CHANGES_REQUESTED** [review]

**Comment by anjneeksharma** [comment]
@diwanshi  I updated the code as per your comment can you plz have a another look.

**Review by diwanshi — APPROVED** [review]

**Pull request merged**